test/integration/pricing/proc_cpuinfo.tap.js

```diff
@@ -5,41 +5,31 @@
 
 'use strict'
 
-const a = require('async')
 const test = require('tap').test
 const glob = require('glob')
-const fs = require('fs')
+const fs = require('fs/promises')
 const parseProcCpuInfo = require('../../../lib/parse-proc-cpuinfo')
 const path = require('path')
 
-test('pricing proc_cpuinfo', function (t) {
+test('pricing proc_cpuinfo', async function (t) {
   const testDir = path.resolve(__dirname, '../../lib/cross_agent_tests/proc_cpuinfo')
-  glob(path.join(testDir, '*.txt'), function globCallback(err, data) {
-    if (err) {
-      throw err
-    }
-    t.ok(data.length > 0, 'should have tests to run')
-    a.each(
-      data,
-      function (name, cb) {
-        runFile(name, cb)
-      },
-      function (err) {
-        t.notOk(err, 'should not have an error')
-        t.end()
-      }
-    )
-  })
-
-  function runFile(name, cb) {
-    fs.readFile(name, function getFile(err, data) {
+  const data = await new Promise((resolve, reject) => {
+    glob(path.join(testDir, '*.txt'), function globCallback(err, fileList) {
       if (err) {
-        throw err
+        return reject(err)
```

Comment:
The tests never go through this code path, do they? I think the current tests you wrote implicitly test this because an error would bubble up, but is it worthwhile to have a more explicit test for not erroring like the replaced code did?

Reply:
This would happen if the the cross-agent test files were moved. It's true, though, that the errors bubble up--I'm not sure that there'd be additional value in testing the fixture file system beyond the errors bubbling up.